core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -461,6 +462,7 @@ public synchronized boolean isReady() {
 
       @Override
       public void cancel(Status reason) {
+        reason = stripCause(reason);
```

Comment:
This one should round-trip on the client. The server should see something like `Status.CANCELLED.withDescription("Client cancelled")`, but the client should be provided back the same cause, since this is used on client-side for some error reporting.